src/functional/program_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -16,24 +16,28 @@
 import dataclasses
 from typing import Any, Final, TypeVar
 
-import numpy as np
-
+import functional.otf.binding.type_specifications as ts
+from functional.iterator.embedded import IndexField
 from functional.otf import languages, stages, step_types, workflow
 from functional.otf.binding import cpp_interface, interface
 from functional.program_processors.codegens.gtfn import gtfn_backend
+from functional.type_system import type_translation
 
 
 T = TypeVar("T")
 
 
-def get_param_description(
-    name: str, obj: Any
-) -> interface.ScalarParameter | interface.BufferParameter:
-    view: np.ndarray = np.asarray(obj)
-    if view.ndim > 0:
-        return interface.BufferParameter(name, tuple(dim.value for dim in obj.axes), view.dtype)
+def get_param_description(name: str, obj: Any) -> interface.Parameter:
+    # TODO: make type_translation extendable and specialize it for the
+    #   bindings to distinguish between buffer-backed fields and index
+    #   fields.
+    if isinstance(obj, IndexField):
+        dtype = type_translation.from_type_hint(obj.dtype.type)
+        assert isinstance(dtype, ts.ScalarType)
+        return interface.Parameter(name, ts.IndexFieldType(axis=obj.axis, dtype=dtype))
```

Comment:
I think, for coherency, this piece of code should be moved into `type_translation.from_value()`, since the similar handling of `embedded.LocatedField` is also implemented there. Additionally, it's easier to refactor the type translation code in the future if it is not spread across different functions.

Reply:
I moved it to a new function `from_value` under `functional.otf.binding.type_translation`. It shouldn't be moved into the main `type_translation` module because `IndexFieldType` is not present there.